plugins/woocommerce/src/Internal/BackInStockNotifications.php

```diff
@@ -0,0 +1,516 @@
+	 * As of WooCommerce 10.0, Back In Stock Notifications will be enabled for 5% of users.
+	 * As of WooCommerce 10.1, Back In Stock Notifications will be enabled for all users.
+	 *
+	 * Feature can be disabled via option `update_option( 'wc_feature_woocommerce_back_in_stock_notifications_enabled', 'no' )`,
+	 * even when this method returns true.
+	 *
+	 * See also \Automattic\WooCommerce\Packages::get_enabled_packages.
+	 *
+	 * @return bool
+	 */
+	public static function is_enabled_for_rollout() {
+		return false;
+	}
+
+	/**
+	 * Returns true if the feature is enabled in this WC instance.
+	 * 
+	 * The user option takes precedence over the rollout period flag.
+	 * 
+	 * Helper method redirected to \Automattic\WooCommerce\Packages\Packages::is_package_enabled.
+	 * 
+	 * @return bool
+	 */
+	public static function is_enabled() {
+		return Packages::is_package_enabled( self::$PACKAGE_NAME );
+	}
+
+	/**
+	 * If WooCommerce Back In Stock Notifications gets activated forcibly, without WooCommerce active
+	 * (e.g. via '--skip-plugins'), remove WooCommerce Back In Stock Notifications initialization functions
+	 * early on in the 'plugins_loaded' timeline.
+	 *
+	 * This is called from \Automattic\WooCommerce\Packages::prepare_packages.
+	 */
+	public static function prepare() {
+
+		// Set flag for activation request to prevent fatal errors when BIS is activated while WC+BIS is already active.
+		$bis_plugin_name = 'woocommerce-back-in-stock-notifications/woocommerce-back-in-stock-notifications.php';
+
+		// This needs to run after the standalone plugin is deactivated to restore the daily task.
+		add_action( 'deactivate_' . $bis_plugin_name, array( __CLASS__, 'maybe_setup_events' ), 20 );
+		
+		// Cleanup events when WooCommerce is deactivated.
+		add_action( 'deactivate_woocommerce/woocommerce.php', array( __CLASS__, 'cleanup_events' ), 20 );
+
+		// Update BIS infrastructure after WooCommerce is installed.
+		add_action( 'woocommerce_installed', array( __CLASS__, 'maybe_update_bis_infrastructure' ), 10, 1 );
+
+		if ( function_exists( 'WC_BIS' ) ) {
+			// This skips the initialization of BIS plugin to avoid duplicate code & fatal errors 
+			// when standalone BIS plugin is active and WC core with BIS merged is loaded.
+			// BIS plugin is then deactivated during plugins_loaded@10 priority from \Automattic\WooCommerce\Packages::on_init().
+			remove_action( 'plugins_loaded', array( WC_BIS(), 'initialize_plugin' ), 9 );
+
+			// When WC_BIS() is present before loading BIS from core, it will fatal during init(), 
+			// so init() needs to be skipped. This should only be triggered once after 
+			// a plugin update during the request when BIS plugin is deactivated.
+			self::$bis_plugin_is_active = true;
+		}		
+
+		// Check for CLI activation via WP-CLI
+		if ( defined('WP_CLI') && WP_CLI ) {
+			global $argv;
+			if ( is_array( $argv ) && in_array( 'plugin', $argv ) && in_array( 'activate', $argv ) && in_array( $bis_plugin_name, $argv ) ) {
```

Comment:
If I want to activate bis via CLI, I'm gonna probably write this:
```
wp plugin activate woocommerce-back-in-stock-notifications
```
Which doesn't include the PHP filename thus it falls to: "Cannot redeclare WC_BIS() etc".

Reply:
Updated.